config/backend-config/namespace_config.go

```diff
@@ -102,6 +103,17 @@ func (nc *namespaceConfig) GetWorkspaceLibrariesForWorkspaceID(workspaceID strin
 	return LibrariesT{}
 }
 
+// GetDataRetentionSettingsForWorkspaceID returns dataRetention settings for workspaceID
+func (nc *namespaceConfig) GetDataRetentionSettingsForWorkspaceID(workspaceID string) DataRetention {
+	nc.mapsMutex.RLock()
+	defer nc.mapsMutex.RUnlock()
+
+	if workspaceDataRetentionSettings, ok := nc.workspaceIDToDataRetentionMap[workspaceID]; ok {
+		return workspaceDataRetentionSettings
+	}
+	return DataRetention{}
```

Comment:
Not sure it's a good idea to just return `DataRetention{}` here. Wouldn't it be more clear to return `(DataRetention, bool)` instead so that we would be sure that the setting isn't there?

If the zero values of the struct are always enough then I guess it can be OK although it might make things a bit more difficult if those defaults were to change. WDYT?